component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal

```diff
@@ -0,0 +1,706 @@
+const string ETH_GET_UNCLE_COUNT_BY_BLOCK_HASH = "eth_getUncleCountByBlockHash";
+const string ETH_GET_UNCLE_COUNT_BY_BLOCK_NUMBER = "eth_getUncleCountByBlockNumber";
+const string ETH_GET_CODE = "eth_getCode";
+const string ETH_SIGN = "eth_sign";
+const string ETH_SEND_TRANSACTION = "eth_sendTransaction";
+const string ETH_SEND_RAW_TRANSACTION = "eth_sendRawTransaction";
+const string ETH_CALL = "eth_call";
+const string ETH_ESTIMATE_GAS = "eth_estimateGas";
+const string ETH_GET_BLOCK_BY_HASH = "eth_getBlockByHash";
+const string ETH_GET_BLOCK_BY_NUMBER = "eth_getBlockByNumber";
+const string ETH_GET_TRANSACTION_BY_HASH = "eth_getTransactionByHash";
+const string ETH_GET_TRANSACTION_BY_BLOCK_HASH_AND_INDEX = "eth_getTransactionByBlockHashAndIndex";
+const string ETH_GET_TRANSACTION_BY_BLOCK_NUMBER_AND_INDEX = "eth_getTransactionByBlockNumberAndIndex";
+const string ETH_GET_TRANSACTION_RECEIPT = "eth_getTransactionReceipt";
+const string ETH_GET_UNCLE_BY_BLOCK_HASH_AND_INDEX = "eth_getUncleByBlockHashAndIndex";
+const string ETH_GET_UNCLE_BY_BLOCK_NUMBER_AND_INDEX = "eth_getUncleByBlockNumberAndIndex";
+const string ETH_GET_COMPILERS = "eth_getCompilers";
+const string ETH_COMPILELLL = "eth_compileLLL";
+const string ETH_COMPILE_SOLIDITY = "eth_compileSolidity";
+const string ETH_COMPILE_SERPENT = "eth_compileSerpent";
+const string ETH_NEW_FILTER = "eth_newFilter";
+const string ETH_NEW_BLOCK_FILTER = "eth_newBlockFilter";
+const string ETH_NEW__PENDING_TRANSACTION_FILTER = "eth_newPendingTransactionFilter";
+const string ETH_UNINSTALL_FILTER = "eth_uninstallFilter";
+const string ETH_GET_FILTER_CHANGES = "eth_getFilterChanges";
+const string ETH_GET_FILTER_LOGS = "eth_getFilterLogs";
+const string ETH_GET_LOGS = "eth_getLogs";
+const string ETH_GET_WORK = "eth_getWork";
+const string ETH_SUBMIT_WORK = "eth_submitWork";
+const string ETH_SUBMIT_HASHRATE = "eth_submitHashrate";
+const string EMPTY_PARAMS = "[]";
+
+@Description {value:"Ethereum client connector"}
+@Param {value:"URI: URI of the Ethereum JSON - RPC server"}
+public connector ClientConnector (string URI, http:Options options, string jsonRPCVersion, int id) {
+
+    endpoint<http:HttpClient> ethereumEP {
+        create http:HttpClient(URI, options);
+    }
+
+    @Description {value:"Returns the current client version"}
+    @Param {value:"jsonRPCVersion: JSON RPC version"}
+    @Param {value:"id: Network id"}
+    @Return {value:"InResponse object"}
+    @Return {value:"Error occured during HTTP client invocation"}
+    action web3ClientVersion () (json, http:HttpConnectorError) {
+        var response, e = ethereumEP.post("/", constructRequest(jsonRPCVersion, id, WEB3_CLIENT_VERSION, EMPTY_PARAMS));
+        return response.getJsonPayload(), e;
+    }
+
+    @Description {value:"Returns Keccak-256 (not the standardized SHA3-256) of the given data"}
+    @Param {value:"jsonRPCVersion: JSON RPC version"}
+    @Param {value:"id: Network id"}
```

Comment:
id and jsonRPCVersion are no longer params for this action. Check all the actions